android/libraries/rib-coroutines/src/main/kotlin/com/uber/rib/core/RibCoroutineScopes.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2022. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.rib.core
+
+import com.uber.autodispose.ScopeProvider
+import com.uber.autodispose.coroutinesinterop.asCoroutineScope
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import java.util.WeakHashMap
+import kotlin.coroutines.CoroutineContext
+import kotlin.reflect.KProperty
+
+val ScopeProvider.coroutineScope: CoroutineScope by LazyCoroutineScope {
```

Comment:
coroutineScope is pretty generic. would it be better to use something more descriptive like `ribScope` like viewmodel uses w/ [viewModelScope](https://developer.android.com/topic/libraries/architecture/coroutines#viewmodelscope)? this would help readability to make it more clear what the scope was

Reply:
Yeah, this is something I've gone back and forth on. It's a scope for anything that implements ScopeProvider, so technically not limited to an interactor or to Ribs.

Here's my thinking:
1. The primary autodispose method is ScopeProvider.asCoroutineScope, so the default one as a property would be ScopeProvider.coroutineScope.
2. I would have preferred ScopeProvider.scope, but it would confuse with Motif.
3. If it was ribScope, then it could be confusing if you had worker.ribScope and interactor.ribScope in the same Rib, as they would imply they are the same, but they aren't the same object.

I also debated if these functions should live in this repo at all, vs in Autodispose since they are ScopeProvider extension functions, but in the end all of this work provides a number of util functions we think will work well with Ribs. We may long term remove the autodispose dependency if we decide to implement native coroutine and lifecycle support in Ribs-base.